### PR TITLE
Fix [Android] Can not set Entry cursor position by tapping if ClearButtonVisibility=WhenEditing

### LINKED ISSUE #7773: [Bug] [Android] Can not set Entry cursor position by tapping if ClearButtonVisibility=WhenEditing
### Description

In Xamarin.Forms Versions prior to V4.3, I was able to set the Cursor position within an Entry by tapping the location I want the cursor to be at.
Since the introduction of `ClearButtonVisibility` this behaviour has changed. If I set `ClearButtonVisibility=WhenEditing` on an entry, I can not change the cursor position by tapping anymore. This happens only if the soft keyboard is open.

### Steps to Reproduce

1. Run the attached solution, you should see two entries there.
2. Enter some random text into the first entry. Then tap anywhere within the entered text - the cursor should then jump to the selected Position accordingly.
3. Enter some random text into the second entry and try to relocate the cursor again. It does not work (the second Entry has `ClearButtonVisibility=WhenEditing` set)

### Expected Behavior

I should be able to set the cursor's position by tapping - even if `ClearButtonVisibility=WhenEditing` is set.

### Actual Behavior

I can only set the cursor's position by tapping if `ClearButtonVisibility=Never` is set (which is the default value).

### Basic Information

- Version with issue: Xamarin.Forms 4.3.0-pre1+
- Last known good version: N/A (ClearButtonVisibility is only available on Version 4.3.0-pre1+)
- IDE: Tested on Visual Studio Professional 2019 (16.3.0)
- Platform Target Frameworks:
  - iOS:  13.1
  - Android: 9.0
  - UWP:  N/A
- Android Support Library Version: 28.0.0.3
- Nuget Packages: Xamarin.Forms & Xamarin.Forms.Visual.Material 4.3.0.819712-pre2
- Affected Devices: Tested on Huawei P Smart + (2018) with Android 8.1 installed

### Reproduction Link

[ClearButtonTest.zip](https://github.com/xamarin/Xamarin.Forms/files/3680039/ClearButtonTest.zip)

### PR DESCRIPTION
**Pull request by melimion**
### Description of Change ###
Fixes touch event handling for EditText when ClearButtonVisibility set to WhileEditing.
Issue was introduced in #5277. Attached touch event handler prevented any other touch actions like cursor position change or context menu (copy/paste) on long tap.

Also fixes boundaries detection of clear button - it can be tested on first Entry with margin in Issue7773. 
Before this PR Entry was cleared when you tap anywhere to the right of text, not only on button.

### Issues Resolved ### 
- fixes #7773

### API Changes ###
 None

### Platforms Affected ### 
- Android

### Behavioral/Visual Changes ###

Touch actions on entry works as expected - cursor position can be set, context menu correctly opens on hold. Entry text cleared only when tapping on clear button.

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###
Control gallery -> Issue7773

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by samhouts** [comment]
@melimion Please rebase to 4.3.0. Thanks!

**Comment by melimion** [comment]
@samhouts done

**Comment by melimion** [comment]
@PureWeen done

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7773.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+					new Entry
+					{
+                        Margin = new Thickness(50),
+						HorizontalOptions = LayoutOptions.FillAndExpand,
+						ClearButtonVisibility = ClearButtonVisibility.WhileEditing,
+						Text = "ClearButtonVisibility"
+					},
+					new Entry
+					{
+                        //Margin = new Thickness(50),
```
[the remark this answers is not in the document]
Comment (melimion) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7773.cs]: @samhouts done

**Pull request merged**